Core 42541 - change getDecisions functionality (new branch)

ninaceban opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description

<!--- Describe your changes in detail -->
Changes related to get and render personalization decisions:
- renamed `viewStart` to `renderDecisions`
- renamed `scopes` to `decisionScopes`
- return decisions at the event command:
- if `renderDecisions` is true we add magical param `__view__` to `decisionScopes`. In the `event::onResponse` we render `dom-actions` decisions items. We return decisions with items that are not `dom-action` in the `event::command` promise.
- if `renderDecisions` is false or not defined and `decisionScopes` contains scopes, we return decisions in the `event:command` promise.

## Related Issue

<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context

<!--- Why is this change required? What problem does it solve? -->

## Screenshots (if appropriate):

## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [ ] I have submitted a [documentation](https://github.com/AdobeDocs/alloy-docs) pull request or no changes are needed.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

jfkhoury commented on src/components/Personalization/createComponent.js:
```diff
@@ -0,0 +1,75 @@
+        const scopes = getDecisionScopes(renderDecisions, decisionScopes);
+
+        if (!hasScopes(scopes)) {
+          logger.info("Personalization is skipped.");
```
I wonder if this is necessary, or if we need more info, like a reason why personalization is skipped. thought?

ninaceban replied:
because `renderDecisions` is false and `decisionScopes` is empty, or missing?

ninaceban replied:
Personalization is skipped because renderDecisions is false and decisionScopes is empty.
like this?

Aaronius replied:
@ninaceban I'm wondering if this message is providing enough value to keep it it around. Is the concern that the user might not know why personalization content isn't being rendered?

jfkhoury commented on src/components/Personalization/createComponent.js:
```diff
@@ -0,0 +1,75 @@
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
```
Love this module now!!

ninaceban replied:
great!

ninaceban commented:
@Aaronius I have addressed all your suggestions and comments from the last PR. 
Regarding `logger.info("Personalization is skipped.");` maybe `logger.debug` would make more sense. cc: @jfkhoury

Aaronius commented:
Ship it!
